### PR TITLE
feat: deployment logs integration test

### PR DESCRIPTION
**Pull request by meniRoy**
Integration tests for logging the latest deployment actions. 
When a user clicks on "redeploy" or selects an environment in the environments view, the latest deployment logs are captured into the VSCode log channel.

Details:
- The tests cover the pulling of logs and their output to the log channel.
- The following scenarios are not covered in this PR:
    - Situations where the step log endpoint returns hasMoreLogs: true.
    - Actions other than "redeploy".
    - Queued deployments.

Please let me know if you'd like additional tests to cover any of the above scenarios or if there are other cases you think should be addressed.

### CODE REVIEW

**Comment by meniRoy** [comment]
[Integration Tests](https://trello.com/c/eoV0MdoJ/10053-integration-tests)

File: src/test/integration/suite/deployment-logs.test.it.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+const auth = { keyId: "key-id", secret: "key-secret" };
+const orgId = "org-id";
+const firstEnvName = "my env";
+const secondEnvName = "my env 2";
+
+let firstEnvironmentMock: EnvironmentModel;
+let secondEnvironmentMock: EnvironmentModel;
+
+suite("deployment logs", function () {
+  this.timeout(1000 * 15);
```
Comment (liranfarage89) [inline src/test/integration/suite/deployment-logs.test.it.ts:38]: what is this timeout?

Comment (meniRoy) [reply src/test/integration/suite/deployment-logs.test.it.ts]: this is the test timeout
3bfa17983c1e965e1addb62032c2981cbfdad006
now its on 10 sec as for all the tests, the reason is that some times it takes a while until the `setInterval` updates the state

File: src/test/integration/suite/deployment-logs.test.it.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    await waitFor(() => expect(outputChannelMock.show).toHaveBeenCalled());
+    await waitFor(() => expect(outputChannelMock.clear).toHaveBeenCalled());
+  });
+
+  test("should show deployment logs when redeploy", async () => {
+    const steps = {
+      "git:clone": ["Cloning into 'repo'..."],
+      "state:get": ["Retrieving persisted state for environment..."],
+    };
+    const newDeploymentId = "new-deployment-id";
```
Comment (sagilaufer1992) [inline src/test/integration/suite/deployment-logs.test.it.ts:125]: this is a default value in many places across this PR. use a different value please, so we know it works as expected, and all values propagate properly

**Review by sagilaufer1992 — APPROVED** [review]
LGTM. left one small comment, if it all works after you deal with it, merge

File: src/test/integration/suite/deployment-logs.test.it.ts (hunk 2 above)
Comment (meniRoy) [reply src/test/integration/suite/deployment-logs.test.it.ts]: a9304ec5ef1e551ca3f176ba7c7ba5127ad1b929

**Pull request merged**